_posts/2020-06-30-hhvm-4.64.markdown

```diff
@@ -0,0 +1,58 @@
+---
+title: "HHVM 4.64"
+layout: post
+author: jjergus
+category: blog
+---
+
+HHVM 4.64 is released! This release marks the end of support for 4.57;
+HHVM 4.58&ndash;4.62 remain supported (4.63 was skipped), as do the 4.32
+and 4.56 LTS releases.
+
+# Breaking Changes
+
+- The INI flag `hhvm.hack_arr_compat_specialization` was removed and its
+  behavior is now enabled by default.
+  This change disables the implicit inter-operability between `varray`s and
+  `darray`s. Instead, `varray`, `darray` and `array` (soon to be removed) are
+  now treated as types in their own right. Specifically:
+  * Using a `varray` where a `darray` is expected (at parameter and return value
+    typehints) throws, and vice versa. The same goes for property typehints,
+    if they are checked at runtime.
+  * `varray`s now maintain a “`vec`-like” layout, with integer keys from 0 to
+    (size - 1). Attempting to set a string key in a `varray` throws, as does
+    attempting to unset any element other than the last one.
+  * Tuples and shapes are internally represented as `varray`s and `darray`s,
+    respectively. As a
+    result, `darray is tuple(...)` returns `false` and `darray as tuple(...)`
+    throws. The same goes for `varray is shape(...)` and
+    `varray as shape(...)`.
+  * Comparing a `varray` and a `darray` using `==` or `===` always returns
+    `false`.
+  * Attempting to compare `varray`s to `darray`s relationally with `<`, `<=`,
```

Comment:
You can still compare two varrays relationally. This is what I said in our internal announcement: "Comparing a darray with a varray or a darray using a relational comparator (>, <, <=, >=, <>, <=>) will now result in an InvalidOperationException."